bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ModelRowAdapter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.hbase.adapters.read;
+
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.hbase.adapters.ResponseAdapter;
+import com.google.cloud.bigtable.hbase.util.ByteStringer;
+import com.google.cloud.bigtable.hbase.util.TimestampConverter;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.KeyValue;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.util.Bytes;
+
+/**
+ * Adapt between a {@link Row} and an hbase client {@link Result}.
+ */
+public class ModelRowAdapter implements ResponseAdapter<Row, Result> {
+
+  @Override
+  public Result adaptResponse(Row response) {
+    if (response == null) {
+      return new Result();
+    }
+
+    SortedSet<Cell> hbaseCells = new TreeSet<>(KeyValue.COMPARATOR);
```

Comment:
This will slow things down.  We need to sort the `Row`'s cells more efficiently.  We should have an algorithm for that somewhere.

Reply:
If that's true, the we should remove the `SortedSet`/`TreeSet` and just use a `List`.

Reply:
We've had users who read rows that have ~1 million cells per row (I'm not sure why...), and this type of operation is costly at that scale.

Reply:
@igorbernstein2 Sorry for confusing PR titles, We implemented [FlatRowAdapter](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/FlatRowAdapter.java#L31) to get FlatRow result using readRow callable. However this change is to create Adapter for convertion of gcj's models.Row to hbase's Result.


@sduskis Thanks for information & suggestion, I would update this to use `List`